### ISSUE TITLE
Drop support for `aws_launch_configuration` 

### ISSUE THREAD

**Issue by bryantbiggs** [terraform-aws-modules/terraform-aws-eks#1679]
## Is your request related to a new offering from AWS?
- No, but yes - https://aws.amazon.com/blogs/compute/amazon-ec2-auto-scaling-will-no-longer-add-support-for-new-ec2-features-to-launch-configurations/

## Is your request related to a problem? Please describe.
- Maintainability

## Describe the solution you'd like.
- As part of the next breaking change, remove support for `aws_launch_configuration` and simplify the module by relying on just `aws_launch_template`

## Describe alternatives you've considered.
- None

## Additional context
Because AWS is not adding any new features, and LT has long superseded LC, I think this will greatly help in simplifying a lot of complexity and duplication within the module

**Comment by bryantbiggs** [comment]
@daroga0002 / @antonbabenko - happy for your thoughts

**Comment by antonbabenko** [comment]
I agree, and it's a good idea to include it in v18.0.0 if possible.

**Comment by bryantbiggs** [comment]
I started on a PR last night and will keep working on it - removing launch configuration, converting from `count` to `for_each`, etc. Its still early but I'll keep working on it over today and tomorrow, stay tuned

**Comment by daroga0002** [comment]
> happy to hear your thoughts

Yup, I think it is without sense to hold launch configurations as launch templates are for long time and we have node groups

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.0.0 :tada:
